### PR TITLE
Add maven support

### PR DESCRIPTION
**Pull request by ejortega**
# Overview
Add parsers for maven and grade projects.

For maven projects
1. Install maven `apt/brew install maven` if not already present
2. Navigate to project and create an effective pom using your project's pom file `mvn help:effective-pom -f pom.xml -Doutput=effective-pom.xml`
3. `phylum analyze effective-pom.xml`

For gradle projects
1. Install gradle if not already present
2. Navigate to project and create lockfile `gradle dependencies --write-locks -q > gradle-dependencies.txt`
3. `phylum analyze gradle-dependencies.txt`

# Checklist
- [x] Does this PR have an associated issue?
- [x] Have you ensured that you have met the expected acceptance criteria?
- [x] Have you created sufficient tests?

# Issue
What issue(s) does this PR close. Use the `closes #<issueNum>` here.

### CODE REVIEW

File: lib/src/lockfiles.rs [hunk 1]
```diff
@@ -165,6 +168,134 @@ impl Parseable for PipFile {
...
+
+impl Parseable for Pom {
+    fn new(filename: &Path) -> Result<Self, io::Error>
+    where
+        Self: Sized,
+    {
+        Ok(Pom(std::fs::read_to_string(filename)?))
+    }
+
+    /// Parses `Pipfile` or `Pipfile.lock` files into a vec of packages
```
Comment (Allan-Clements) [inline lib/src/lockfiles.rs:195]: Copy paste comment?

**Review by Allan-Clements — COMMENTED** [review]
LGTM but should get an approval from someone who has worked in the CLI tool previously too.

File: lib/src/lockfiles.rs (hunk 1 above)
Comment (ejortega) [reply lib/src/lockfiles.rs]: Yes it was. Updated.
Thanks!

File: lib/src/lockfiles.rs [hunk 2]
```diff
@@ -165,6 +168,134 @@ impl Parseable for PipFile {
...
+
+impl Parseable for Pom {
+    fn new(filename: &Path) -> Result<Self, io::Error>
+    where
+        Self: Sized,
+    {
+        Ok(Pom(std::fs::read_to_string(filename)?))
+    }
+
+    /// Parses maven effecti-pom files into a vec of packages
```
Comment (Allan-Clements) [inline lib/src/lockfiles.rs:195]:
```suggestion
    /// Parses maven effective-pom files into a vec of packages
```

**Pull request merged**